packages/babel-traverse/src/path/family.ts

```diff
@@ -312,6 +312,53 @@ export function getAllPrevSiblings(this: NodePath): NodePath[] {
   return siblings;
 }
 
+type digitalWithoutZero = "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9";
+type digital = "0" | digitalWithoutZero;
+
+// only support 0 to 999 (since we won't use too large number in path.get())
+type Num =
+  | `${digital}`
+  | `${digitalWithoutZero}${digital}`
+  | `${digitalWithoutZero}${digital}${digital}`;
+
+// convert "1" to 0 (string index to number index)
+type ToNumber<T extends string> = T extends Num ? 0 : T;
+
+type Pattern<Obj extends string, Prop extends string> = `${Obj}.${Prop}`;
```

Comment:
Q: Does TS make any guarantee about which dot it will split the pattern at?

Reply:
Sure. The left most one.

Reply:
Infect, even if TS doesn't make guarantee about which dot it will split, we could change our `Split` to

```ts
type Split<P extends string> = P extends Pattern<infer O, infer U>
  ? [...Split<O>, ...Split<U>] // recursive on both O and U
  : [MaybeToIndex<P>];
```

to make it work as expected!